pkgs/top-level/all-packages.nix

```diff
@@ -6543,6 +6543,7 @@ with pkgs;
   cudaPackages_11 = cudaPackages_11_7;
 
   cudaPackages_12_0 = callPackage ./cuda-packages.nix { cudaVersion = "12.0"; };
+  cudaPackages_12_1 = callPackage ./cuda-packages.nix { cudaVersion = "12.1"; };
   cudaPackages_12 = cudaPackages_12_0;
```

Comment:
Still waiting on a version of CUDNN for CUDA 12.1, so I didn't want to make it the default since that's a large component of the ecosystem.

I don't know that we have any written guidance on changing default versions if it would cause breakages (e.g. anything building with CUDA 12 which requires CUDNN would no longer compile).